trompace/mutations/__init__.py

```diff
@@ -13,6 +14,20 @@ def __repr__(self):
         return self.value
 
 
+class _Neo4jDate(StringConstant):
+    """The _Neo4jDate is used for Date values. It will be added as
+    StringConstant in the GraphQL. The date will be formatted
+    as: { year:[int] [month: [int] day: [int]] }
+
+    The constructor argument can be a datetime or a year."""
+
+    def __init__(self, value):
+        if isinstance(value, datetime):
+            self.value = "{{ year: {0} month: {1} day: {2} }}".format(value.year, value.month, value.day)
```

Comment:
we need to be careful as to what values we want to support here - we could say that a birth/death date could be represented as a `datetime.date` (not datetime). In MusicBrainz, this value can be yyyy, or yyyy-mm or yyyy-mm-dd. I don't know if the python type can represent just a yyyy or yyyy-mm.

In the case that we want to represent a `created` field for an object, this should also contain the hour, minute, second values, We should also make sure that this kind of conversion properly handles this case (and add tests).

We had some discussion about how to represent dates for which we didn't have exact values (e.g. "around 1710", "born on the 27th or 28th"). We haven't talked about this in much more detail, so I think that we can skip it for now, but we should think of a nice way to represent it in this client too.

Reply:
I agree that the `datetime.date` type has the right precision for the field. A python date type must contain a valid date, so with month and day included. Regarding support for year-month support we can consider checking for a list type as value. The values should contain [year, month, day] where day and month are optional.

The uncertain dates can not be represented by the current _Neo4jDate type. In Schema.org the date type is defined by the ISO 8601 standards where in the basic rules there is also no support, but in the [extensions](https://www.iso.org/standard/70908.html) there is.

Reply:
because the CE is currently using the _Neo4jDate type, let's not focus on uncertain dates. We can come back to this in the future if we find a clear need for it.

Reply:
Added support for the suggested year+month support. Also included tests for the _Neo4jDate type in test_mutations.py. Please verify the solution.